File: domain/src/main/java/home/oleg/placenearme/models/DetailedVenue.java

package home.oleg.placenearme.models;

import java.util.Collections;
import java.util.List;

public class DetailedVenue {

    private Category category;
    private Contact contact;
    private String description;
    private Hours hours;
    private String id;
    private Location location;
    private String name;
    private List<Photo> photos;
    private boolean isFavorite;
    private float rating;
    private double distance;

    public double getDistance() {
        return distance;
    }

    public void setDistance(double distance) {
        this.distance = distance;
    }

    public Contact getContact() {
        return contact;
    }

    public void setContact(Contact contact) {
        this.contact = contact;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public boolean isFavorite() {
        return isFavorite;
    }

    public void setFavorite(boolean favorite) {
        isFavorite = favorite;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Hours getHours() {
        return hours;
    }

    public void setHours(Hours hours) {
        this.hours = hours;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Photo> getPhotos() {
        if(photos == null) {
            return Collections.emptyList();
        }
        return Collections.unmodifiableList(photos);
    }

    public void setPhotos(List<Photo> photos) {
        this.photos = photos;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }

}

File: core-network/src/main/java/home/oleg/placesnearme/network/models/DetailedVenue.kt
package home.oleg.placesnearme.network.models

import java.util.ArrayList
import java.util.Collections

import io.reactivex.annotations.NonNull

data class DetailedVenue(
        val canonicalUrl: String? = null,
        val categories: List<Category>? = null,
        val contact: Contact? = null,
        val createdAt: Long? = null,
        val description: String? = null,
        val hours: Hours? = null,
        val id: String,
        val likes: Likes? = null,
        val location: Location,
        val name: String,
        val photoGroup: PhotosGroup? = null,
        val rating: Float = 0.toFloat(),
        val shortUrl: String? = null,
        val stats: Stats? = null,
        val storeId: String? = null,
        val timeZone: String? = null,
        val url: String? = null) {

    val photos: List<Photo>
        @NonNull
        get() {
            if (photoGroup!!.groups == null) {
                return emptyList()
            }

            val res = ArrayList<Photo>()

            for (photoGroup in photoGroup.groups) {
                res.addAll(photoGroup.items)
            }

            return res
        }

}
